src/group_data.cpp

```diff
@@ -1,11 +1,13 @@
 #include "dplyr.h"
 
-SEXP dplyr_group_indices(SEXP rows, SEXP s_nr) {
-  R_xlen_t nr = INTEGER(s_nr)[0];
-  R_xlen_t ng = XLENGTH(rows);
+SEXP dplyr_group_indices(SEXP data, SEXP rows) {
+  R_xlen_t nr = vctrs::short_vec_size(data);
+  if (nr == 0) return dplyr::vectors::empty_int_vector;
```

Comment:
Control flow on own line so it's visible:

```suggestion
  if (nr == 0) {
    return dplyr::vectors::empty_int_vector;
  }
```